hopr/hopr-lib/src/lib.rs

```diff
@@ -13,6 +13,9 @@
 //! For most of the practical use cases, the `hoprd` application should be a preferable
 //! choice.
 
+#[cfg(all(feature = "runtime-async-std", feature = "runtime-tokio"))]
+compile_error!("Only one of the runtime features can be specified for the build");
```

Comment:
Case #1 will result in build failures. And we cannot chose one at this point, since we don't know whether `runtime-tokio` actually works. Most crates with multiexecutor support do not force a choice though.